File: 1838-frequency-of-the-most-frequent-element/1838-frequency-of-the-most-frequent-element.java
class Solution {
    public int maxFrequency(int[] nums, int k) {
        Arrays.sort(nums);
        
        int l = 0, r = 0;
        long sum = 0;
        int ans = 0;
        
        while (r < nums.length) {
        	sum += nums[r];
        	while ((nums[r] * (r - l + 1)) > (sum + k)) {
        		sum -= nums[l];
        		l++;
        	}
        	
        	ans = Math.max(ans, (r - l + 1));
        	r++;
        }
        
        return ans;
    }
}

File: 1838-frequency-of-the-most-frequent-element/1838-frequency-of-the-most-frequent-element.py
class Solution:
    def maxFrequency(self, nums: List[int], k: int) -> int:
        nums.sort()
        length = len(nums)
        left = res = 0
        windowSum = 0
        
        for i in range(length):
            windowSum += nums[i]
            maxVal = nums[i]
            
            while (windowSum + k) < (maxVal * (i - left + 1)):
                windowSum -= nums[left]
                left += 1
            
            res = max(res, i - left + 1)
        
        return res
            
                